### PR TITLE
New quickstart template to demonstrate transitive BGP connections + fixed "Deploy to Azure" link for vNet Peering template

### PR DESCRIPTION
**Pull request by bhummerstone**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
1-00: New template

### Description of the change
Created new template to demonstrate transitive BGP routing between multiple vNets

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@bhummerstone__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (benhu). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: 201-vnet-transitive-bgp/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "sharedKey": {
+            "value": "mysharedkey1234"
```
Comment (bmoore-msft) [inline 201-vnet-transitive-bgp/azuredeploy.parameters.json]: replace this with GEN-UNIQUE to prevent accidental deployments with this value...

Comment (bhummerstone) [inline 201-vnet-transitive-bgp/azuredeploy.parameters.json]: I've tried using GEN-UNIQUE before, and it failed the CI build due to the sharedKey only accepting certain characters

Comment (bmoore-msft) [inline 201-vnet-transitive-bgp/azuredeploy.parameters.json]: really, gen-unique should be alphanumeric (gen-password has non-alpha chars)... can you try it and if it fails we'll have a log (for a bug) and then we can rollback the change...

**Comment by bhummerstone** [comment]
Sorted - perhaps I was using gen-password before :)

**Pull request merged**